File: ios/SsdidWallet/Domain/Verifier/Offline/UnifiedVerificationResult.swift
import Foundation

enum VerificationStatus {
    case verified
    case verifiedOffline
    case failed
    case degraded
}

enum VerificationSource {
    case online
    case offline
}

enum CheckType {
    case signature
    case expiry
    case revocation
    case bundleFreshness
}

enum CheckStatus {
    case pass
    case fail
    case unknown
}

struct VerificationCheck {
    let type: CheckType
    let status: CheckStatus
    let message: String
}

struct UnifiedVerificationResult {
    let status: VerificationStatus
    let checks: [VerificationCheck]
    let source: VerificationSource
    var bundleAge: TimeInterval? = nil
}

File: android/sdk/ssdid-core/src/main/java/my/ssdid/sdk/domain/verifier/offline/UnifiedVerificationResult.kt
package my.ssdid.sdk.domain.verifier.offline

import java.time.Duration

data class UnifiedVerificationResult(
    val status: VerificationStatus,
    val checks: List<VerificationCheck>,
    val source: VerificationSource,
    val bundleAge: Duration? = null
)

enum class VerificationStatus { VERIFIED, VERIFIED_OFFLINE, FAILED, DEGRADED }
enum class VerificationSource { ONLINE, OFFLINE }

data class VerificationCheck(
    val type: CheckType,
    val status: CheckStatus,
    val message: String
)

enum class CheckType { SIGNATURE, EXPIRY, REVOCATION, BUNDLE_FRESHNESS }
enum class CheckStatus { PASS, FAIL, UNKNOWN }
